docs/guides/api-graphql/fragments/android/subscriptions-by-id.md

```diff
@@ -0,0 +1,124 @@
+            + "content "
+            + "id "
+            + "postCommentsId "
+        + "}"
+    + "}";
+    return new SimpleGraphQLRequest<>(
+            document,
+            Collections.singletonMap("id", id),
+            Comment.class,
+            new GsonVariablesSerializer());
+}
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+private fun onCreateCommentByPostId(id: String): GraphQLRequest<Comment?>? {
+    val document = ("subscription onCreateCommentByPostId(\$id: ID!) { "
+        + "onCommentByPostId(postCommentsId: \$id) { "
+            + "content "
+            + "id "
+            + "postCommentsId "
+        + "}"
+    + "}")
+    return SimpleGraphQLRequest(
+            document,
+            mapOf("id", id),
+            Comment::class.java,
+            GsonVariablesSerializer())
+}
+```
+
+</amplify-block>
+</amplify-block-switcher>
+
+To listen to creation updates with the specific post using the post id, you can use the following code sample:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+private void createSubscription() {
+    Amplify.API.subscribe(onCreateCommentByPostId("12345"),
+            subscriptionId -> Log.d("MyAmplifyApp", "Established subscription with id: " + subscriptionId), 
+            response -> {
+                if(response.hasErrors()) {
+                    Log.e("MyAmplifyApp", "Error receiving Comment: " + response.getErrors());
+                } else if(!response.hasData()) {
+                    Log.e("MyAmplifyApp", "Error receiving Comment, no data in response.");
+                } else {
+                    Log.d("MyAmplifyApp", "Successfully got comment from subscription: " + response.getData());
+                }
+            }, 
+            exception -> Log.e("MyAmplifyApp", "Subscription terminated with error: " + exception), 
+            () -> Log.d("MyAmplifyApp", "Subscription has been closed successfully."));
+}
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+private fun createSubscription() {
+    Amplify.API.subscribe(onCreateCommentByPostId("12345")!!,
+            { subscriptionId: String -> Log.d("MyAmplifyApp", "Established subscription with id: $subscriptionId") },
+            { response: GraphQLResponse<Comment?> ->
+                if (response.hasErrors()) {
+                    Log.e("MyAmplifyApp", "Error receiving Comment: " + response.errors)
+                } else if (!response.hasData()) {
+                    Log.e("MyAmplifyApp", "Error receiving Comment, no data in response.")
+                } else {
+                    Log.d("MyAmplifyApp", "Successfully got comment from subscription: " + response.data)
+                }
+            },
+            { exception: ApiException -> Log.e("MyAmplifyApp", "Subscription terminated with error: $exception") }
+    ) { Log.d("MyAmplifyApp", "Subscription has been closed successfully.") }
+}
```

Comment:
Suggest a few changes:
1. Various strategies to reduce line lengths, to optimize for a broad range of (possibly constrained) viewports:
    1. Use the `$it` and `it` shorthand for lambda arguments
    2. Reduce verbosity of log messages
    3. Use +4 indentation instead of +8
2. Bring the outer closure into the call. Having it out-fixed makes it look special in some way. But really, it is more like the other closures and having them behave the same reaffirms their logical grouping.

Open question -- can we get rid of the `!!` and the `?` on the `Comment?` ?
```suggestion
private fun createSubscription() {
    Amplify.API.subscribe(onCreateCommentByPostId("12345")!!,
        { Log.d("MyAmplifyApp", "Established subscription. id = $it") },
        { response: GraphQLResponse<Comment?> ->
            if (response.hasErrors()) {
                Log.e("MyAmplifyApp", "Error receiving Comment: " + response.errors)
            } else if (!response.hasData()) {
                Log.e("MyAmplifyApp", "Error receiving Comment; no data in response.")
            } else {
                Log.d("MyAmplifyApp", "Got comment on subscription: " + response.data)
            }
        },
        { Log.e("MyAmplifyApp", "Subscription terminated with error.", it) },
        { Log.d("MyAmplifyApp", "Subscription has been closed successfully.") }
    )
}
```

Reply:
> Open question -- can we get rid of the `!!` and the `?` on the `Comment?` ?

Yep, done!